### PR TITLE
Windows: Add missing HostConfig fields to update block

### LINKED ISSUE #31575: --cpus parameter can not be updated with Windows Server containers and docker inspect alwasy shows CpuCount equals 0
Hello there,

I am running my test "microsoft/iis" container with following command with "--cpus 16" limitation:
```
docker run -d  --name gt_test_iis --cpus 16 microsoft/iis ping -t localhost
```

Looks like "--cpus" works fine and when I run my `simulate_cpu_load.ps1` script:
```
ForEach ($Number in 1..16){
   start-job -ScriptBlock{
      $result = 1; foreach ($number in 1..2147483647) {$result = $result * $number}
   }
}
```

I got only about 16% CPU on my 64 cpus virtual machine. 

When I wanted to change CPU limits online I tried `docker update` and got expected error (as --cpus flag is not described there https://docs.docker.com/engine/reference/commandline/update/#options):
```
PS> docker update --cpus 1 gt_test_iis
unknown flag: --cpus
See 'docker update --help'.
```

But on the API documentation page for "docker update" ( https://docs.docker.com/engine/api/v1.26/#operation/ContainerUpdate ) we have following option `CpuCount`, so from my understanding "docker update" should have "--cpus" parameter.

Another thing I found is that changing the value of `CpuCount` by Web API does not affect running container.

When I tried to invoke it in Powershell console:

```
Invoke-WebRequest -Uri http://hostname:2375/containers/gt_test_iis/update -Method POST -Body '{"CpuCount":1}' -ContentType 'application/json'
```
it returned 200
```
StatusCode        : 200
StatusDescription : OK
Content           : {"Warnings":[]}

RawContent        : HTTP/1.1 200 OK
                    Api-Version: 1.26
                    Docker-Experimental: false
                    Content-Length: 16
                    Content-Type: application/json
                    Date: Mon, 06 Mar 2017 12:32:40 GMT
                    Server: Docker/1.13.1-cs1 (windows)

                    {"Warnin...
Forms             : {}
Headers           : {[Api-Version, 1.26], [Docker-Experimental, false], [Content-Length, 16], [Content-Type, application/json]...}
Images            : {}
InputFields       : {}
Links             : {}
ParsedHtml        : mshtml.HTMLDocumentClass
RawContentLength  : 16
```

but the container continued to consume 16% of CPU load.

I also found that "docker inspect" always shows `"CpuCount": 0,` even it was run with "--cpus 16" or without "--cpus" at all, and it was not changed when I run `Invoke-WebRequest` above.

**Output of `docker version`:**

```
Client:
 Version:      1.13.1-cs1
 API version:  1.26
 Go version:   go1.7.5
 Git commit:   8709b81
 Built:        Thu Feb  9 02:05:36 2017
 OS/Arch:      windows/amd64

Server:
 Version:      1.13.1-cs1
 API version:  1.26 (minimum version 1.24)
 Go version:   go1.7.5
 Git commit:   8709b81
 Built:        Thu Feb  9 02:05:36 2017
 OS/Arch:      windows/amd64
 Experimental: false
```

**Output of `docker info`:**

```
Containers: 248
 Running: 111
 Paused: 0
 Stopped: 137
Images: 1201
Server Version: 1.13.1-cs1
Storage Driver: windowsfilter
 Windows:
Logging Driver: json-file
Plugins:
 Volume: local
 Network: l2bridge l2tunnel nat null overlay transparent
Swarm: inactive
Default Isolation: process
Kernel Version: 10.0 14393 (14393.693.amd64fre.rs1_release.161220-1747)
Operating System: Windows Server 2016 Datacenter
OSType: windows
Architecture: x86_64
CPUs: 64
Total Memory: 1.906 TiB
Name: skipped
ID: 62EZ:P3MP:SQ3U:3UET:3JAP:QLAU:X2O4:E5NA:7WAH:BUGN:CAQT:VZ5V
Docker Root Dir: E:\docker_storage_1_13
Debug Mode (client): false
Debug Mode (server): true
 File Descriptors: -1
 Goroutines: 721
 System Time: 2017-03-06T13:49:13.9906709Z
 EventsListeners: 1
Registry: https://index.docker.io/v1/
Experimental: false
Insecure Registries:
 127.0.0.0/8
Live Restore Enabled: false
```

**Additional environment details (AWS, VirtualBox, physical, etc.):**
It is x1.x32large AWS instance. But the issue was reproduced on all our Docker 1.13.1-cs1 installations.

### PR DESCRIPTION
**Pull request by darstahl**
Signed-off-by: Darren Stahl <darst@microsoft.com>

**- What I did**

Fixes #31575 

Some values (CpuCount and CpuPercent, among others) were not being checked during container update. This resulted in returning success when using the Update Container API with resources set to be updated, even though they were not updated.

**- How I did it**

Ensure all fields in the Resources field of HostConfig are not set when updating containers on Windows daemons.

**- How to verify it**

See https://github.com/docker/docker/issues/31575 for repro steps:

Proof of fix:
image

**- Description for the changelog**

Block resource updating on Windows containers instead of silently failing in some cases

/cc @jhowardmsft @jstarks FYI

### CODE REVIEW

File: container/container_windows.go
Comment (jhowardmsft) [inline container/container_windows.go]: Nit - lowercase `r` on `Resource`

Comment (darrenstahlmsft) [inline container/container_windows.go]: fixed

**Comment by jhowardmsft** [comment]
LGTM. Ping @johnstep

**Comment by MHBauer** [comment]
@jhowardmsft @jstarks @johnstep Is there any reason to hold off on merging this?

**Comment by jhowardmsft** [comment]
@MHBauer No - just needs another maintainers LGTM :smile:

**Review by cpuguy83 — APPROVED** [review]
LGTM

**Pull request merged**
